src/server.ts

```diff
@@ -1,8 +1,73 @@
 import express, { Request, Response } from 'express';
 
+import { Prisma, PrismaClient, User } from '@prisma/client';
+
+const prisma = new PrismaClient();
+
 export const server = express();
+
+type ErrorMessage = {
+  message: string;
+};
+
+type EmptyObj = Record<string, never>;
+
+// Middlewares
+server.use(express.json());
+
+// Endpoints
 server.get('/', (_: Request, res: Response) => {
   res.send({
     message: 'hello world',
   });
 });
+
+// User endpoints
+server.get(
+  '/users',
+  async (
+    _: Request<EmptyObj, User[], EmptyObj>,
+    res: Response<User[]>
+  ): Promise<Response<User[]>> => {
+    const users = await prisma.user.findMany();
+    return res.send(users);
+  }
+);
+
+interface NewUserRequestBodyParams {
+  uid: string;
+  username: string;
+}
+
+server.post(
+  '/users/new_user',
+  async (
+    req: Request<EmptyObj, User, NewUserRequestBodyParams>,
+    res: Response<User | ErrorMessage>
+  ): Promise<Response<User | ErrorMessage>> => {
+    const { username, uid } = req.body;
+    if (!username || !uid)
+      return res
+        .status(400)
+        .send({ message: 'Need both a username and a uid' });
+    // TODO: add unit tests
+    // TODO: add auth0 layer
+    try {
+      const user = await prisma.user.create({ data: { username, uid } });
```

Comment:
this try/catch block is very much needed to catch the case of an existing primary key constraint.  If we don't include it, the response from the express server waits indefinitely.